src/main/scala/com/fulcrumgenomics/bam/IdentifyPrimers.scala

```diff
@@ -0,0 +1,1027 @@
+ @arg(          doc="Maximum mismatches for a primer to be considered a match.") val maxMismatches: Int = 1,
+ @arg(          doc="The minimum alignment score for fully gapped alignment for a primer to be considered a match.") val minAlignmentScore: Int = 10,
+ @arg(          doc="The match score to use for aligning to primer sequences.") val matchScore: Int = 1,
+ @arg(          doc="The mismatch score to use for aligning to primer sequences.") val mismatchScore: Int = -4,
+ @arg(          doc="The gap open score to use for aligning to primer sequences.") val gapOpen: Int = -6,
+ @arg(          doc="The gap extension score to use for aligning to primer sequences.") val gapExtend: Int = -1,
+ @arg(          doc="The minimum difference in mismatches between the best match and second best match to accept a primer match.  Not used for full alignment.") val minMismatchDelta: Int = 1,
+ @arg(flag='t', doc="The number of threads to use.") val threads: Int = 1,
+ @arg(flag='n', doc="Examine the first N templates.") val numTemplates: Option[Int] = None,
+ @arg(          doc="Skip full-alignment matching") val skipFullAlignment: Boolean = false,
+ @arg(          doc="Path to the ksw aligner.") val ksw: Option[String] = None,
+ @arg(          doc="The maximum number of templates in memory.") val maxTemplatesInRam: Option[Int] = None,
+ @arg(          doc="The number of templates to process at a time per thread.") val templatesPerThread: Int = 1000
+) extends FgBioTool with LazyLogging {
+
+  import IdentifyPrimers._
+
+  private val kswExecutable: Option[FilePath] = this.ksw.map(Paths.get(_)).flatMap {
+    case p if Files.exists(p) => Some(p)
+    case name =>
+      val path = System.getenv("PATH")
+      validate(path != null, "PATH environment variable was not found; required for --ksw")
+      val systemPath = path.split(File.pathSeparatorChar).view.map(PathUtil.pathTo(_))
+      systemPath.map(p => p.resolve(name)).find(ex => Files.exists(ex))
+        .orElse {
+          throw new ValidationException(s"Could not find ksw executable ${ksw} in PATH: $path")
+        }
+  }
+
+  Io.assertReadable(input)
+  Io.assertCanWriteFile(output)
+  Io.assertCanWriteFile(metrics)
+  kswExecutable.foreach(Io.assertReadable)
+
+  validate(maxMismatches >= 0,   "--max-mismatches must be >= 0")
+  validate(minMismatchDelta > 0, "--min-mismatch-delta must be > 0")
+  validate(matchScore > 0,       "--match-score must be > 0")
+  validate(mismatchScore < 0,    "--mismatch-score must be < 0")
+  validate(gapOpen < 0,          "--gap-open must be < 0")
+  validate(gapExtend < 0,        "--gap-extend must be < 0")
+  maxTemplatesInRam.foreach { m => validate(templatesPerThread < m, "--max-templates-in-ram must be greater than or equal to --templates-per-thread")}
+
+  private val aligner: Aligner = Aligner(matchScore = matchScore, mismatchScore = mismatchScore, gapOpen = gapOpen, gapExtend = gapExtend, mode = AlignmentMode.Glocal)
```

Comment:
Is this just used to get the final alignment with cigar etc?  If so could you comment that here please?